### PR TITLE
mgmt: mcumgr: callbacks: Fix issue with OR'ing events

### LINKED ISSUE #53549: mgmt: mcumgr: callbacks: Callbacks events for a single group should be able to be combined
**Describe the bug**
There is an issue in the callback notification function, it _should_ allow for events that are part of a single group to be OR'd together (but not multiple groups), however it seems that this is not implemented and it will only allow the specific event or all events for a single group or all events in total.

In addition, the documentation (and sample, and test code) can be updated to improve how this is described to users and tested.

**To Reproduce**
Combine 2 events in the same group

**Expected behavior**
The callback should work for both

**Impact**
Medium

**Environment (please complete the following information):**
 - Commit SHA or Version used: a326f05b93cbaa4927eddca90036fb263c0e54f7

### PR DESCRIPTION
**Pull request by nordicjm**
This fixes an issue whereby callback event IDs in the same group cannot be combined. It adds a new test to ensure that this is working and clarifies the documentation to improve understanding of what is and is not allowed, including examples or good and bad usage.

Fixes #53549

### CODE REVIEW

**Comment by nordicjm** [comment]
CC @mariuszpos @Damian-Nordic

File: doc/services/device_mgmt/mcumgr_callbacks.rst [hunk 1]
```diff
@@ -78,6 +78,28 @@ setup a callback for an os_mgmt callback, this must be done as a separate
 registration. Group IDs are numerical increments, event IDs are bitmask values,
 hence the restriction.
 
+As an example, the following is registration is allowed, which will register
```
Comment (mbolivar-nordic) [inline doc/services/device_mgmt/mcumgr_callbacks.rst:81]: `following registration is`

**Review by mbolivar-nordic — DISMISSED** [review]

File: doc/services/device_mgmt/mcumgr_callbacks.rst (hunk 1 above)
Comment (nordicjm) [reply doc/services/device_mgmt/mcumgr_callbacks.rst]: Good catch, fixed.

**Review by de-nordic — DISMISSED** [review]

File: subsys/mgmt/mcumgr/mgmt/src/mgmt.c [hunk 2]
```diff
@@ -106,9 +106,12 @@ int32_t mgmt_callback_notify(uint32_t event, void *data, size_t data_size)
 		struct mgmt_callback *loop_group =
 			CONTAINER_OF(snp, struct mgmt_callback, node);
 
-		if (loop_group->event_id == event || loop_group->event_id == MGMT_EVT_OP_ALL ||
+		if (loop_group->event_id == MGMT_EVT_OP_ALL ||
 		    (MGMT_EVT_GET_GROUP(loop_group->event_id) == group &&
-		     MGMT_EVT_GET_ID(loop_group->event_id) == MGMT_EVT_OP_ID_ALL)) {
+		     (MGMT_EVT_GET_ID(loop_group->event_id) == MGMT_EVT_OP_ID_ALL ||
```
Comment (Damian-Nordic) [inline subsys/mgmt/mcumgr/mgmt/src/mgmt.c:111]: This line is probably no longer needed now as with `MGMT_EVT_OP_ID_ALL = 0xffff` it reduces to the second condition below.

**Review by Damian-Nordic — DISMISSED** [review]

File: subsys/mgmt/mcumgr/mgmt/src/mgmt.c (hunk 2 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/mgmt/src/mgmt.c]: Fixes.

**Review by Damian-Nordic — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**